test/testDeployCli.js

```diff
@@ -165,29 +165,32 @@ Authentication is required. You can pass the key via the HLX_WSK_AUTH environmen
     process.env.HLX_FOOBAR = '1234';
     new CLI()
       .withCommandExecutor('deploy', mockDeploy)
-      .run(['deploy', '--no-auto',
+      .run(['deploy',
         '--wsk-auth', 'secret-key',
         '--wsk-namespace', 'hlx',
       ]);
   });
 
-  it('hlx deploy can disable auto', () => {
+  it.skip('hlx deploy can enable auto', (done) => {
     new CLI()
       .withCommandExecutor('deploy', mockDeploy)
-      .run(['deploy',
+      .onFail((err) => {
+        assert.equal(err, 'Error: Auto-deployment requires: --circleci-auth, --fastly-auth, --fastly-namespace');
+        done();
+      })
+      .run(['deploy', '--auto', 'true',
         '--wsk-auth', 'secret-key',
         '--wsk-namespace', 'hlx',
-        '--no-auto',
       ]);
 
-    sinon.assert.calledWith(mockDeploy.withEnableAuto, false);
+    sinon.assert.calledWith(mockDeploy.withEnableAuto, true);
```

Comment:
those checks are executed too early. you need to either move them into the `onFail()`  case, or even better:
- change the test to be `async` (instead of done)
- `await` for the run
- add a try/catch around it, and check the error there
- add the asserts after the try/catch

Reply:
seems correct